distribution/lib/Standard/Base/0.0.0-dev/src/Random.enso

```diff
@@ -1,99 +1,307 @@
-   replacement.
+    ## PRIVATE
+    Value (java_random:Java_Random)
 
-   If the amount of elements to select is larger than the input vector size, it
-   returns a random permutation of the input vector.
-sample : Vector Any -> Integer -> Random_Number_Generator -> Vector Any
-sample vector k rng =
-    new_array = Random_Utils.sample vector k rng.java_random
-    Vector.from_polyglot_array new_array
+    integer : Integer -> Integer -> Integer ! Illegal_Argument
+    integer self (min:Integer) (max:Integer) =
+        range = max - min
+        if range >= Java_Integer.MAX_VALUE then Error.throw (Illegal_Argument.Error "Currently only integer ranges of up to 2^31-1 are supported.") else
+          min + (self.java_random.nextInt range)
 
-## PRIVATE
-   Returns `k` indices sampled from the range [0, n-1] without replacement.
+    float : Float -> Float -> Float
+    float self (min:Float=0.0) (max:Float=1.0) =
+        range = max - min
+        self.java_random.nextDouble * range + min
 
-   If `k >= n`, it will return a random permutation of the indices.
-random_indices : Integer -> Integer -> Random_Number_Generator -> Vector Integer
-random_indices n k rng =
-    array = Random_Utils.random_indices n k rng.java_random
-    Vector.from_polyglot_array array
+    boolean : Boolean
+    boolean self =
+        self.java_random.nextBoolean
+
+    gaussian : Float -> Float -> Float
+    gaussian self (standard_deviation:Float=1.0) (mean:Float=0.0) =
+        self.java_random.nextGaussian * standard_deviation + mean
+
+    indices : Integer -> Integer -> Vector Integer
+    indices self (n:Integer) (count:Integer=1) =
+        array = Random_Utils.random_indices n count self.java_random
+        Vector.from_polyglot_array array
+
+    date : Date_Range -> Date
+    date self (date_range:Date_Range) = date_range.at (self.integer 0 date_range.length)
+
+    time : Time_Of_Day -> Time_Of_Day -> Time_Of_Day
+    time self (start_time:Time_Of_Day=(Time_Of_Day.new 0 0 0)) (end_time:Time_Of_Day=(Time_Of_Day.new 23 59 59)) =
+        ## The time range end is inclusive because it is impossible to specify
+           an end time after the last second of the day.
+        seconds = self.integer 0 (end_time.to_seconds - start_time.to_seconds + 1)
+        start_time.date_add seconds Time_Period.Second
+
+    uuid : Text
+    uuid self = UUID.randomUUID.to_text
+
+    item : Vector Any -> Integer -> Boolean -> Any
+    item self (v:Vector) (count:Integer=1) (with_replacement:Boolean=False) =
+        case with_replacement of
+            True ->
+                len = v.length
+                0.up_to 1 . map _-> v.at (self.integer 0 len)
+            False ->
+                new_array = Random_Utils.sample v count self.java_random
+                Vector.from_polyglot_array new_array
+
+    permute : Vector Any -> Vector Any
+    permute self (v:Vector) = self.item v v.length False
```

Comment:
Once the dust settles on this new version, and we make the instances available, I'll copy all the documentation from the static methods to the instance methods.

Reply:
Do you plan to do this on this PR or a next one?

For now these methods are not exposed to the users, so IMO it would make sense to annotate them as `PRIVATE`. We can always change that later - un-privating a method is imagine the 'least breaking' change.

Reply:
This will be a later PR. Added PRIVATE.